lib/mllp/client.ex

```diff
@@ -104,13 +164,36 @@ defmodule MLLP.Client do
 
   def format_error(err), do: inspect(err)
 
+  @doc """
+  Starts a new MLLP.Client.
+
+  MLLP.Client.start_link/4 will start a new MLLP.Client process. i
+
+  This function will raise a `RuntimeError` if an invalid `ip_address()` is provided. 
+
+  ## Options
+
+  * `:reply_timeout` - Optionally specify a timeout value for receiving a response. Must be a positive integer or 
+     `:infinity`. Defaults to `:infinity`.
+    
+  * `:socket_opts` -  A list of socket options as supported by [`:gen_tcp`](`:gen_tcp`). 
+     Note that `:binary`, `:packet`, and `:active` can not be overriden.
+
+  * `:telemetry_module` - A callback module for which MLLP.Client will call `YourTelmetryModule.exectute/3`. The 
```

Comment:
Puposefully  did  not document this in  detail as the current telemetry parameters and usage overall needs polish and revision. As  an example,  for metadata we pass internal state to the callback in whole

Reply:
I think, we don't need to include telemetry_module if we just emit some meaningful telemetry events using telemetry.span or execute.

Reply:
I agree with this. I was simply documented what was there. I will undocument this and open an issue or PR to remove the option.